ofrak_core/ofrak/gui/server.py

```diff
@@ -739,6 +742,61 @@ async def search_for_vaddr(self, request: Request) -> Response:
         except NotFoundError:
             return json_response([])
 
+    @exceptions_to_http(SerializedError)
+    async def search_for_string(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        string_query_string = body["search_query"]
+        regex = body["regex"]
+        case_ignore = body["case_ignore"]
+        if not isinstance(string_query_string, str):
+            raise ValueError("Invalid search query.")
+        string_query: Union[bytes, re.Pattern[bytes]] = string_query_string.encode()
+        if case_ignore:
+            string_query = re.compile(string_query, re.IGNORECASE)
```

Comment:
This annoyed me, but I couldn't find an alternate way. The only way I could see to efficiently do case insensitivity is with regex, so if someone uses a regex pattern it will still work even if not checked. So I made them mutually inclusive, and when you check `Ignore Case` in the GUI it will auto check the regex option as well.

Reply:
One solution is to use `.lower()` (or `.casefold()`) on the string before running the query. But that's definitely less efficient than a regular expression. 

To continue using the regular expression, you should probably [run `re.escape` on the `string_query` first(https://docs.python.org/3/library/re.html#re.escape).

```suggestion
            string_query = re.compile(re.escape(string_query), re.IGNORECASE)
```